examples/cudax/simple_p2p/simple_p2p.cu

```diff
@@ -0,0 +1,253 @@
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *  * Neither the name of NVIDIA CORPORATION nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * This sample demonstrates a combination of Peer-to-Peer (P2P) and
+ * Unified Virtual Address Space (UVA) features.
+ */
+
+#include <cuda/memory_resource>
+
+#include <cuda/experimental/algorithm.cuh>
+#include <cuda/experimental/buffer.cuh>
+#include <cuda/experimental/device.cuh>
+#include <cuda/experimental/launch.cuh>
+#include <cuda/experimental/memory_resource.cuh>
+
+#include <algorithm>
+
+#include <stdio.h>
+#include <stdlib.h>
+
+namespace cudax = cuda::experimental;
+
+struct simple_kernel
+{
+  template <typename Dimensions>
+  __device__ void operator()(Dimensions dims, ::cuda::std::span<const float> src, ::cuda::std::span<float> dst)
+  {
+    // Just a dummy kernel, doing enough for us to verify that everything worked
+    const auto idx = dims.rank(cudax::thread);
+    dst[idx]       = src[idx] * 2.0f;
+  }
+};
+
+std::vector<cudax::device_ref> find_peers_group()
+{
+  // Check possibility for peer access
+  printf("\nChecking GPU(s) for support of peer to peer memory access...\n");
+
+  std::vector<cudax::device_ref> peers;
+  for (auto& dev_i : cudax::devices)
+  {
+    for (auto& dev_j : cudax::devices)
+    {
+      if (dev_i != dev_j)
+      {
+        bool can_access_peer = dev_i.is_peer_accessible_from(dev_j);
+        // Save all peers of a first device found with a peer
+        if (can_access_peer && peers.size() == 0)
+        {
+          peers = dev_i.get_peers();
+          peers.insert(peers.begin(), dev_i);
+        }
+        printf("> Peer access from %s (GPU%d) -> %s (GPU%d) : %s\n",
+               dev_i.get_name().c_str(),
+               dev_i.get(),
+               dev_j.get_name().c_str(),
+               dev_j.get(),
+               can_access_peer ? "Yes" : "No");
+      }
+    }
+  }
+
+  return peers;
+}
```

Comment:
I believe this could reuse `device_ref::get_peers`

Reply:
It is using `get_peers` in line 75 to get the actual peers.

The original sample was printing out for every pair of devices if peer access is available, so I thought I should do it too. So I added the printing loop to show how you can access individual pairs, but also used `get_peers` as a way to construct the resulting vector to show that API as well.
But if this is too obscure, I could also get rid of the printing loop or separate it out

Reply:
definitely separate it out. i'll also note that the printing loops are doing a cross-product, so this could be made linear with a `cartesian_product` view. but i don't think we have such a thing yet.